subsys/net/lib/sockets/sockets.c

```diff
@@ -2530,6 +2581,30 @@ static int sock_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 		return 0;
 	}
 
+	case ZFD_IOCTL_FIONBIO:
+		sock_set_flag(obj, SOCK_NONBLOCK, SOCK_NONBLOCK);
+		return 0;
+
+	case ZFD_IOCTL_FIONREAD: {
+		int *avail;
+		ssize_t len;
+		struct net_pkt *pkt;
+		struct net_context *ctx = (struct net_context *)obj;
+
+		avail = va_arg(args, int *);
+		pkt = k_fifo_peek_head(&ctx->recv_q);
+		if (pkt == NULL) {
+			*avail = 0;
+			return 0;
+		}
```

Comment:
Should we have a loop here that checks all the pkt in the receive queue, I think we discussed this in networking forum?

Reply:
Yes, thanks for the reminder. I was debugging TLS and forgot to add that testcase.

Reply:
Loop added! I don't think it's quite perfect yet, but getting there.